### PR TITLE
Ignore empty attrs

### PR DESCRIPTION
**Pull request by SogoKato**
According to [slog-handler-guide](https://github.com/golang/example/blob/master/slog-handler-guide/README.md#the-handle-method), empty attributes should be ignored.

> Next, it follows the handler rule that says that empty attributes should be ignored.

## Test

using `example/example.go`

```go
	// ...
	logger := slog.New(slogslack.Option{
		Level:      slog.LevelDebug,
		WebhookURL: webhook,
		Channel:    channel,
		ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
			if a.Key == "some_hidden_key" {
				return slog.Attr{}
			}
			return a
		},
	}.NewSlackHandler())
	// ...
	logger.
		// ...
		With("some_hidden_key", "secret").
		Error("A message")
```

### Before

Add an attribute `some_hidden_key` and modify with `ReplaceAttr` then it will print `<nil>`.

screenshot 1

### After

Now it prints nothing for `some_hidden_key`.

screenshot 2

### CODE REVIEW

**Comment by samber** [comment]
Hi @SogoKato and thanks for your suggestion.

I think we should implement that in slog-common.

File: converter.go [hunk 1]
```diff
@@ -38,6 +38,10 @@ func DefaultConverter(addSource bool, replaceAttr func(groups []string, a slog.A
 func attrToSlackMessage(base string, attrs []slog.Attr, message *slack.WebhookMessage) {
 	for i := range attrs {
 		attr := attrs[i]
+		if attr.Equal(slog.Attr{}) {
```
Comment (samber) [inline converter.go:41]: @SogoKato i wonder why it works, `attr` is expected to have at least a `Key`.

image

Should it be `attr.Equal(slog.Attr{Key: attr.Key})` instead ?

Comment (samber) [reply converter.go:41]: Or `attr.Value.Equal(slog.Value{Kind: attr.Value.Kind})`

**Comment by SogoKato** [comment]
> Hi @SogoKato and thanks for your suggestion.
> 
> I think we should implement that in slog-common.

Hello @samber !

I will try to work on this.

File: converter.go (hunk 1 above)
Comment (SogoKato) [reply converter.go:41]: When `slog.Attr{}`, the `Key` is empty string (which is initial value) so it works.

```go
log.Println("slog.Attr{}.Key == \"\"", slog.Attr{}.Key == "")
```

then

```
2024/07/01 18:08:11 slog.Attr{}.Key == "" true
```

**Comment by samber** [comment]
@SogoKato Can you review it please ?

https://github.com/samber/slog-common/pull/6

**Comment by SogoKato** [comment]
> @SogoKato Can you review it please ?
> 
> [samber/slog-common#6](https://github.com/samber/slog-common/pull/6)

@samber  
Thank you for creating a p-r. I've reviewed that and left a comment.

And, I've tested by importing slog-common locally (by adding a line `replace github.com/samber/slog-common => ../../../github.com/samber/slog-common`), and found it works!

https://github.com/samber/slog-slack/blob/15370ee606485fcdfdd7e8ed05af557976b1cb2e/converter.go#L22

```go
// added this line
attrs = slogcommon.RemoveEmptyAttrs(attrs)
```

image

**Comment by SogoKato** [comment]
@samber Thank you for releasing v0.17.0 of slog-common. I reverted the previous commit and pushed new commits. Could you review it again?

**Comment by samber** [comment]
I just made the update on slog-slack -> v2.6.0

**Pull request closed without merging**